src/screens/claimables/shared/components/ClaimButton.tsx

```diff
@@ -0,0 +1,68 @@
+import { AccentColorProvider, Box, Inline, Text, TextShadow } from '@/design-system';
+import { deviceUtils, watchingAlert } from '@/utils';
+import React from 'react';
+import { ButtonPressAnimation, ShimmerAnimation } from '@/components/animations';
+import { debounce } from 'lodash';
+import { useWallets } from '@/hooks';
+import { enableActionsOnReadOnlyWallet } from '@/config';
+
+const BUTTON_WIDTH = deviceUtils.dimensions.width - 52;
+
+export function ClaimButton({
+  onPress,
+  disabled,
+  shimmer,
+  biometricIcon,
+  label,
+}: {
+  onPress: () => void;
+  disabled: boolean;
+  shimmer: boolean;
+  biometricIcon: boolean;
+  label: string;
+}) {
+  const { isReadOnlyWallet } = useWallets();
+
+  return (
+    <ButtonPressAnimation
+      disabled={disabled}
+      style={{ width: '100%', paddingHorizontal: 18 }}
+      scaleTo={0.96}
+      onPress={debounce(() => {
```

Comment:
why debounce here?

Reply:
since state updates are async it is possible for user to spam click and trigger multiple executions of onPress before button can be disabled